Add sensu-backend-init for Docker and remove notes (#2124)
sensu/sensu-docs: content/sensu-go/5.17/installation/install-sensu.md

### 3. Initialize

**With the backend running**, run `sensu-backend init` to set up your Sensu administrator username and password.
In this initialization step, you only need to set environment variables with a username and password string &mdash; no need for role-based access control (RBAC).

Replace `YOUR_USERNAME` and `YOUR_PASSWORD` with the username and password you want to use.

{{< language-toggle >}}

{{< highlight Docker >}}
docker run \
-e SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=YOUR_USERNAME \

> -e SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=`YOUR_PASSWORD' \

Author: ```suggestion
-e SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=YOUR_PASSWORD \
```

--restart always \
--name sensu \
sensu/sensu:latest \

> sensu-backend start \

Author: Should `sensu-backend start \` be removed here?

Reviewer: It needs to stay - we don't need line 162: `sensu-backend init` as it's called in the Dockerfile with some logic to only run if the container is being launched as a backend.

https://github.com/sensu/sensu-enterprise-go/blob/fix/dockerfile-init-user-settable-admin/docker-scripts/sensu-entrypoint.sh#L21

Reviewer 2: Paging @amdprophet

sensu-backend init
{{< /highlight >}}

{{< highlight "Docker Compose" >}}
---

> NEEDED

Author: ```suggestion
---
version: "3"
services:
  sensu-backend:
    image: sensu/sensu:latest
    ports:
    - 3000:3000
    - 8080:8080
    - 8081:8081
    volumes:
    - "sensu-backend-data:/var/lib/sensu/etcd"
    command: "sensu-backend-init -e SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=YOUR_USERNAME -e SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=YOUR_PASSWORD"
volumes:
  sensu-backend-data:
    driver: local
```

### 3. Initialize

**With the backend running**, run `sensu-backend init` to set up your Sensu administrator username and password.
In this initialization step, you only need to set environment variables with a username and password string &mdash; no need for role-based access control (RBAC).

Replace `YOUR_USERNAME` and `YOUR_PASSWORD` with the username and password you want to use.

{{< language-toggle >}}

{{< highlight Docker >}}
docker run \
-e SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=YOUR_USERNAME \
-e SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=YOUR_PASSWORD \
--restart always \
--name sensu \
sensu/sensu:latest \

> sensu-backend start \

Reviewer: Remove the `\` here.

> sensu-backend start \

Author: ```suggestion
sensu-backend start
```

> sensu-backend init

Reviewer: This line can be removed.

> sensu-backend init

Author: ```suggestion
```

{{< /highlight >}}

{{< highlight "Docker Compose" >}}
---
version: "3"
services:
  sensu-backend:
    image: sensu/sensu:latest
    ports:
    - 3000:3000
    - 8080:8080
    - 8081:8081
    volumes:
    - "sensu-backend-data:/var/lib/sensu/etcd"

>     command: "sensu-backend-init -e SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=YOUR_USERNAME -e SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=YOUR_PASSWORD"

Reviewer: This should just be `command: "sensu-backend start"` I believe. On a new line set the environment variables like:

```
    environment:
      - SENSU_BACKEND_CLUSTER_ADMIN_USERNAME
      - SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD
```

>     command: "sensu-backend-init -e SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=YOUR_USERNAME -e SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=YOUR_PASSWORD"

Reviewer 2: Hmm, not sure this makes sense, as sensu-backend init is not a service, but rather a one time command to be run.

Author: I don't know -- I guessed on the Compose placeholder here based on the command. Do @amdprophet 's changes straighten it out?

>     command: "sensu-backend-init -e SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=YOUR_USERNAME -e SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=YOUR_PASSWORD"

Author: ```suggestion
    command: "sensu-backend start"
    environment:
    - SENSU_BACKEND_CLUSTER_ADMIN_USERNAME
    - SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD
```

volumes:
  sensu-backend-data:
    driver: local
{{< /highlight >}}

{{< highlight "Ubuntu/Debian" >}}
export SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=YOUR_USERNAME
export SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=YOUR_PASSWORD

> sensu-backend init

Reviewer: Change `sensu-backend init` to `docker-compose up`. The instructions are the same for Ubuntu & RHEL; can probably get rid of the platform highlight here.

Author: I'm sorry, I think I'm misunderstanding this one -- These instructions are already in the doc for Ubuntu/Debian (see the existing [step 3](https://docs.sensu.io/sensu-go/latest/installation/install-sensu/#3-initialize)).
Because we're adding Docker instructions now, I added the tab and platform format we use in the docs in step 3. So it will appear on an Ubuntu/Debian tab like we have in [step 2](https://docs.sensu.io/sensu-go/latest/installation/install-sensu/#2-configure-and-start). They are the same as for RHEL/CentOS, but we use a platform switcher that separates the two.

Should we have docker instructions in the Ubuntu/Debian tab? Or do you mean we should add `docker-compose-up` at the end of the Docker Compose code?

Reviewer: My bad - you can ignore this entire suggestion.

### 3. Initialize

**With the backend running**, run `sensu-backend init` to set up your Sensu administrator username and password.
In this initialization step, you only need to set environment variables with a username and password string &mdash; no need for role-based access control (RBAC).

Replace `YOUR_USERNAME` and `YOUR_PASSWORD` with the username and password you want to use.

{{< language-toggle >}}

{{< highlight Docker >}}
docker run \
-e SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=YOUR_USERNAME \
-e SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=YOUR_PASSWORD \
--restart always \
--name sensu \
sensu/sensu:latest \
sensu-backend start \
sensu-backend init
{{< /highlight >}}

{{< highlight "Docker Compose" >}}

> ---

Author: ```suggestion
```

### 3. Initialize

**With the backend running**, run `sensu-backend init` to set up your Sensu administrator username and password.
In this initialization step, you only need to set environment variables with a username and password string &mdash; no need for role-based access control (RBAC).

Replace `YOUR_USERNAME` and `YOUR_PASSWORD` with the username and password you want to use.

{{< language-toggle >}}

{{< highlight Docker >}}
docker run \
-e SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=YOUR_USERNAME \
-e SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=YOUR_PASSWORD \
--restart always \
--name sensu \
sensu/sensu:latest \
sensu-backend start
{{< /highlight >}}

{{< highlight "Docker Compose" >}}
---
version: "3"
services:
  sensu-backend:
    image: sensu/sensu:latest
    ports:
    - 3000:3000
    - 8080:8080
    - 8081:8081
    volumes:
    - "sensu-backend-data:/var/lib/sensu/etcd"
    command: "sensu-backend start"
    environment:

>     - SENSU_BACKEND_CLUSTER_ADMIN_USERNAME

Reviewer: I mistakingly thought these env vars were already set. We'll have to add `=YOUR_USERNAME` and `=YOUR_PASSWORD` to the end of these two lines.

>     - SENSU_BACKEND_CLUSTER_ADMIN_USERNAME

Author: ```suggestion
    - SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=YOUR_USERNAME
```

>     - SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD

Author: ```suggestion
    - SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=YOUR_PASSWORD
```

After starting the Sensu backend, open the web UI by visiting http://localhost:3000.
You may need to replace `localhost` with the hostname or IP address where the Sensu backend is running.

> To log in to the web UI, enter your Sensu user credentials (the user ID and password you provided with `sensu-backend init`).

Reviewer: Rather than mention `sensu-backend init`, which runs behind the scenes, it may be better to say something like:

(the user ID and password provided with the `SENSU_BACKEND_CLUSTER_ADMIN_USERNAME` and `SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD` environment variables).

> To log in to the web UI, enter your Sensu user credentials (the user ID and password you provided with `sensu-backend init`).

Author: ```suggestion
To log in to the web UI, enter your Sensu user credentials (the user ID and password you provided with the (the user ID and password provided with the `SENSU_BACKEND_CLUSTER_ADMIN_USERNAME` and `SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD` environment variables).
```

After starting the Sensu backend, open the web UI by visiting http://localhost:3000.
You may need to replace `localhost` with the hostname or IP address where the Sensu backend is running.

> To log in to the web UI, enter your Sensu user credentials (the user ID and password you provided with the (the user ID and password provided with the `SENSU_BACKEND_CLUSTER_ADMIN_USERNAME` and `SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD` environment variables).

Author: ```suggestion
To log in to the web UI, enter your Sensu user credentials (the username and password provided with the `SENSU_BACKEND_CLUSTER_ADMIN_USERNAME` and `SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD` environment variables).
```
